pkg/http/api.go

```diff
@@ -307,18 +313,75 @@ func (a *api) onOutputBindingMessage(reqCtx *fasthttp.RequestCtx) {
 	}
 }
 
-func (a *api) onGetState(reqCtx *fasthttp.RequestCtx) {
+func (a *api) onBulkGetState(reqCtx *fasthttp.RequestCtx) {
+	store, err := a.getStateStoreWithRequestValidation(reqCtx)
+	if err != nil {
+		log.Debug(err)
+		return
+	}
+
+	var req BulkGetRequest
+	err = a.json.Unmarshal(reqCtx.PostBody(), &req)
+	if err != nil {
+		msg := NewErrorResponse("ERR_MALFORMED_REQUEST", err.Error())
+		respondWithError(reqCtx, 400, msg)
+		return
+	}
+
+	metadata := getMetadataFromRequest(reqCtx)
+
+	bulkResp := []BulkGetResponse{}
+	var wg sync.WaitGroup
+
+	for _, k := range req.Keys {
+		wg.Add(1)
+		go func(k string) {
+			defer wg.Done()
+
+			gr := &state.GetRequest{
+				Key:      a.getModifiedStateKey(k),
+				Metadata: metadata,
+			}
+
+			resp, err := store.Get(gr)
+			if err != nil {
+				log.Debugf("bulk get: error getting key %s: %s", k, err)
+			} else if resp != nil && resp.Data != nil {
+				bulkResp = append(bulkResp, BulkGetResponse{
+					Key:  k,
+					Data: jsoniter.RawMessage(resp.Data),
+					ETag: resp.ETag,
+				})
+			}
+		}(k)
+	}
+	wg.Wait()
+
+	b, _ := a.json.Marshal(bulkResp)
+	respondWithJSON(reqCtx, 200, b)
```

Comment:
There is no limit on # of keys, what happens when the data to be returned for all keys is greater than the http response message size here, how does user knows about it?

Reply:
We don't have response message size limits.

Reply:
But there is a limit, its not limitless :)

Reply:
We don't put message size limits. The same is true for getting a single item that has it's size over any possible message limits. so.. this is not special to bulk of items or a single item.

In both cases, the user will get an error saying the message size was exceeded.

Reply:
This is special to bulk items, as users are more likely to get into this situation with bulk get. There is no good way, dapr can handle this as sizes will be varying across different items and not all components will support paging. So this error seems fine